app/redux/localNode/actions.js

```diff
@@ -92,3 +91,28 @@ export const getUpcomingEarnings = (): Action => async (dispatch: Dispatch): Dis
     throw err;
   }
 };
+
+export const setLocalNodeStorage = ({ capacity, drive }: { capacity: { id: number, label: string }, drive: { id: number, label: string } }): Action => async (
+  dispatch: Dispatch
+): Dispatch => {
+  try {
+    const commitmentSize = Math.floor(capacity.id / 1048576); // Bytes to MB
+    const logicalDrive = drive.label;
+    await httpService.setCommitmentSize({ commitmentSize });
+    await httpService.setLogicalDrive({ logicalDrive });
+    dispatch({ type: SET_ALLOCATION, payload: { capacity, drive } });
+  } catch (err) {
+    dispatch({ type: SET_ALLOCATION, payload: { capacity: null, drive: null } });
```

Comment:
no need to reset state because this call is also used to update settings and we don't want to reset previous settings

Reply:
ok